src/VisualStudio/CSharp/Impl/CSharpPackage.cs

```diff
@@ -111,11 +119,13 @@ private void RegisterObjectBrowserLibraryManager()
             }
         }
 
-        private void UnregisterObjectBrowserLibraryManager()
+        private async Task UnregisterObjectBrowserLibraryManagerAsync(CancellationToken cancellationToken)
         {
             if (_libraryManagerCookie != 0)
             {
-                if (this.GetService(typeof(SVsObjectManager)) is IVsObjectManager2 objectManager)
+                await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
hmm.. so this we switch before QI but not in counter part method?

Reply:
➡️ This was unintentional, and was fixed in 7b7c87da83e465ed489a1aa5de856bb3fca701d3.